src/app/inventory/selectors.ts

```diff
@@ -106,25 +109,97 @@ export const materialsSelector = (state: RootState) =>
 /** The actual raw profile response from the Bungie.net profile API */
 export const profileResponseSelector = (state: RootState) => state.inventory.profileResponse;
 
-/** A set containing all the hashes of owned items. */
-export const ownedItemsSelector = createSelector(
-  profileResponseSelector,
-  allItemsSelector,
-  (profileResponse, allItems) => {
-    const ownedItemHashes = new Set<number>();
-    for (const item of allItems) {
-      ownedItemHashes.add(item.hash);
+const STORE_SPECIFIC_OWNERSHIP_BUCKETS = [
+  // Emblems cannot be transferred between characters and if one character owns an emblem,
+  // other characters don't really own it. Also affects vendor claimability.
+  BucketHashes.Emblems,
+  // Quests and bounties are character-specific.
+  BucketHashes.Quests,
+];
+
+/**
+ * Sets of items considered "owned" for checkmark purposes, some
+ * account-scoped, some character-scoped.
+ *
+ * Most items are considered owned from the view of a character if
+ * they're in any bucket because they can be transferred or are in
+ * the consumables bucket, but for quests and bounties, it's necessary
+ * to see whether the current character has them.
+ */
+export interface OwnedItemsInfo {
+  accountWideOwned: Set<number>;
+  storeSpecificOwned: {
+    [key: string]: Set<number>;
+  };
+}
+
+/**
+ * Sets containing all the hashes of owned items, globally and from the
+ * view of individual characters. Excludes plugs, see
+ * ownedUncollectiblePlugsSelector for those.
+ */
+export const ownedItemsSelector = createSelector(allItemsSelector, (allItems) => {
+  const accountWideOwned = new Set<number>();
+  const storeSpecificOwned = {};
+  for (const item of allItems) {
+    if (STORE_SPECIFIC_OWNERSHIP_BUCKETS.includes(item.bucket.hash)) {
+      if (!storeSpecificOwned[item.owner]) {
+        storeSpecificOwned[item.owner] = new Set();
+      }
+      storeSpecificOwned[item.owner].add(item.hash);
```

Comment:
Potentially distasteful from a readability perspective, but:
```suggestion
      (storeSpecificOwned[item.owner] ??= new Set()).add(item.hash);
```